ocaml/typing/includecore.ml

```diff
@@ -700,6 +705,31 @@ module Record_diffing = struct
     else
       Some (diffing loc env params1 params2 l r)
 
+  let find_mismatch_in_mixed_record_representations
+      ({ value_prefix_len = v1; flat_suffix = s1 } : mixed_product_shape)
+      ({ value_prefix_len = v2; flat_suffix = s2 } : mixed_product_shape)
+    =
+    if v1 = v2 then None
+    else
+      (* It's currently possible for the value prefix length to legally differ
```

Comment:
Yes.... but this is really OK? Is there a test showing so? It seems very weird to me to have two objects of the same type floating about with different headers in this way. But I don't actually see yet what goes wrong. Does marshalling? Does polymorphic comparison?

Reply:
> Does marshalling? Does polymorphic comparison?

Chris and I talked about this: Both of these are unsupported for mixed blocks (i.e. will raise). So we think it's fine. I'll add a test.

My one concern is the optimizer. I'm discussing with flambda2 devs whether this causes issues. If it does, I'll keep immediates in the value prefix. But we should tackle that separately from this PR.

Reply:
(Will open a follow-up PR for this.)